docker/Dockerfile

```diff
@@ -13,12 +15,17 @@ RUN pip3 install python_magic
 RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 99
 RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 99
 
-RUN useradd --create-home --shell /bin/bash user
-COPY . /home/user/gg/
-RUN chown user -R /home/user/gg/
+RUN mkdir /app
+COPY . /app/gg
 
 ENV LANG C.UTF-8
 ENV LANGUAGE C:en
 ENV LC_ALL C.UTF-8
 
-USER user
+WORKDIR /app/gg
+
+# dev-env stage
+FROM build-env as dev-env
+
+RUN pip3 install boto3 numpy python_magic
+RUN apt-get install -y -q vim wget unzip cmake ffmpeg
```

Comment:
> Could we split this into two Dockerfiles? Like so: https://stackoverflow.com/a/53660229/461597

I think splitting into two Dockerfiles is useful when the source Dockerfile gets too large. We can understand this from the original question - "Is it possible to split this one big-ish Dockerfile into multiple files?" (c) https://stackoverflow.com/q/53659993

> The ffmpeg should not be part of the docker container except the one container used for testing the example.

> Or, alternatively, the name could remain the same .travis.Dockerfile to make it clear that this Dockerfile is for travis only.

1. First of all, I want to unify Travis and local (or working) Docker's setups to make sure that test (Travis) build works the same as local. Therefore, I added a separate `build-env` stage that includes the minimum number of dependencies required to build\compile (only) `gg` framework.

2. Secondly, I want to make an environment where anyone could play with `gg` and run any example for examples folder. Sort of sandbox. And yes, I understand your thoughts, and initially, I wanted to make several Docker builds individual for each example.

> I am foreseeing that there will also be one container for GCC based examples.

But after long thinking, I realized that this approach has many disadvantages:
- building for each example of its container take time and place.
- if you want to run many examples, you cannot do it in one shell, and you have to switch between containers (and remember which one you launched).
- documentation is tricky. You cannot read just example's Docs, and you have to "combine" it with Docker's docs.

=> difficult to use, especially if you are new to docker :( 

One-containers-for-all-examples has only one con: a mess of dependencies. And I agree with you that it could be confusing. What do you think if we rename `dev-env` to `sandbox-env` and add comments to `Dockerfile` like:

```sh
# sandbox-env

# ggfunctions deps
RUN pip3 install boto3 numpy python_magic

# common deps
RUN apt-get install -y -q vim wget unzip

# used for excamera and viddec examples
RUN apt-get install -y -q ffmpeg
```

Reply:
> Additionally, we could also mention that people should build their own development container with build-env as the base.

Good idea, I like it!